plugins/modules/ec2_ami.py

```diff
@@ -501,9 +501,9 @@ def create_image(module, connection):
         module.fail_json_aws(e, msg="Error registering image")
 
     if wait:
+        delay = 15
+        max_attempts = wait_timeout // delay
         waiter = connection.get_waiter('image_available')
-        delay = wait_timeout // 30
-        max_attempts = 30
```

Comment:
The get_waiter call has been switched to use one of our helpers (so that it can better handle RateLimit errors) but takes the same arguments as before.

I'd still really like to get your additional change in here with the update of the timeout/delay logic.  If you rebase then we can get this merged.